pt_miniscreen/root.py

```diff
@@ -152,90 +127,56 @@ def _wait_for_bootsplash_finish(self):
         # start showing main app
         self.state.update({"show_bootsplash": False})
 
-    def _get_upper_icon_path(self):
-        if self.can_exit_menu:
-            return get_image_file_path("gutter/left_arrow.png")
-
-        if self.stack.active_component.can_scroll_up():
-            return get_image_file_path("gutter/top_arrow.png")
-
-    def _get_lower_icon_path(self):
-        if self.can_perform_action:
-            return get_image_file_path("gutter/tick.png")
-
-        if self.can_start_project:
-            return get_image_file_path("gutter/play.png")
-
-        if self.can_enter_menu:
-            return get_image_file_path("gutter/right_arrow.png")
-
     def _set_gutter_icons(self):
-        self.right_gutter.state.update(
-            {
-                "upper_icon_path": self._get_upper_icon_path(),
-                "lower_icon_path": self._get_lower_icon_path(),
-            }
-        )
+        if isinstance(self.active_component, HasGutterIcons):
+            self.right_gutter.state.update(
+                {
+                    "upper_icon_path": self.active_component.top_gutter_icon(
+                        self.stack
+                    ),
+                    "lower_icon_path": self.active_component.bottom_gutter_icon(),
+                }
+            )
 
-    def should_animate_stack_operation(self):
-        return not isinstance(self.active_page, MenuList)
+    @property
+    def is_actionable(self):
+        return isinstance(self.active_page, Actionable)
 
-    def enter_selected_row(self):
-        if self.can_enter_menu:
-            self.stack.push(
-                self.active_page.child,
-                animate=self.should_animate_stack_operation(),
-            )
-            self._set_gutter_icons()
-
-            if self.is_project_page:
-                self.active_page.run(on_stop=self.exit_menu)
-
-    def enter_menu(self):
-        if not self.can_enter_menu:
-            return
-        if isinstance(self.active_page, MenuPage):
-            self.stack.push(self.active_page.PageList)
-            self._set_gutter_icons()
-        elif isinstance(self.active_page, MenuList):
-            self.enter_selected_row()
-
-    def exit_menu(self):
-        if self.can_exit_menu:
-            self.stack.pop(animate=self.should_animate_stack_operation())
-            self._set_gutter_icons()
-
-    def scroll_up(self):
-        if self.can_scroll:
-            self.stack.active_component.scroll_up()
-            self._set_gutter_icons()
-
-    def scroll_down(self):
-        if self.can_scroll:
-            self.stack.active_component.scroll_down()
-            self._set_gutter_icons()
-
-    def select_next_row(self):
-        if self.can_select_row:
-            self.active_page.select_next_row()
-
-    def select_previous_row(self):
-        if self.can_select_row:
-            self.active_page.select_previous_row()
+    @property
+    def handles_button_events(self):
```

Comment:
```suggestion
    def active_component_handles_button_events(self):
```
the fact that this was an @property, combined with it's name not mentioning the active component led to me being confused and thinking there was some important boolean as to whether the root is handling buttons or not.